six/modules/c++/cphd/source/CPHDWriter.cpp

```diff
@@ -164,13 +164,23 @@ void CPHDWriter::addImage<std::complex<float> >(
         const types::RowCol<size_t>& dims,
         const sys::ubyte* vbmData);
 
-void CPHDWriter::writeMetadata(size_t vmbSize,
-                               size_t cphdSize)
+void CPHDWriter::writeMetadata(size_t vbmSize,
+                               size_t cphdSize,
+                               const std::string& classification,
+                               const std::string& releaseInfo)
 {
     const std::string xmlMetadata(CPHDXMLControl().toXMLString(mMetadata));
 
     FileHeader header;
-    header.set(xmlMetadata.size(), vmbSize, cphdSize);
+    header.set(xmlMetadata.size(), vbmSize, cphdSize);
+    if (classification != "")
```

Comment:
Use `if (!classification.empty())` instead (same throughout other spots in code)